src/test/groovy/hudson/plugins/gradle/injection/MavenExtensionsHandlerTest.groovy

```diff
@@ -0,0 +1,68 @@
+package hudson.plugins.gradle.injection
+
+import hudson.FilePath
+import hudson.plugins.gradle.injection.MavenExtensionsHandler.MavenExtension
+import org.junit.Rule
+import org.junit.rules.TemporaryFolder
+import spock.lang.Specification
+import spock.lang.Subject
+import spock.lang.Unroll
+
+class MavenExtensionsHandlerTest extends Specification {
+
+    @Rule
+    public TemporaryFolder tempFolder = new TemporaryFolder()
+
+    @Subject
+    MavenExtensionsHandler mavenExtensionsHandler = new MavenExtensionsHandler()
+
+    @Unroll
+    def "only copies extension if it doesn't exist"() {
+        given:
+        def folder = tempFolder.newFolder()
+        def root = new FilePath(folder)
+
+        when:
+        def firstFilePath = mavenExtensionsHandler.copyExtensionToAgent(mavenExtension, root)
+
+        then:
+        firstFilePath.exists()
+        def firstLastModified = firstFilePath.lastModified()
+        firstLastModified > 0
+
+        when:
+        def secondFilePath = mavenExtensionsHandler.copyExtensionToAgent(mavenExtension, root)
+
+        then:
+        secondFilePath.exists()
+        secondFilePath.remote == firstFilePath.remote
+        def secondLastModified = firstFilePath.lastModified()
+        secondLastModified == firstLastModified
+
+        where:
+        mavenExtension << MavenExtension.values().toList()
```

Comment:
not sure you need the `toList` here ?

Reply:
It works without `toList`. The only difference is that type inference is not working properly in this case:

<img width="1012" alt="Screenshot 2022-09-20 at 11 56 48" src="https://user-images.githubusercontent.com/1210272/191228246-5ebea5a2-f91c-4a10-b090-16460039307f.png">

Will leave `toList` to not to have warning in the IDE

Reply:
all right. You could specify the `mavenExtension` + its type in the parameter of the test method, to not rely on the inference and specify what the type should be, but I'm ok keeping as it is

Reply:
It's a great idea! Added parameter and dropped `toList`